src/XRP/default-xrp-client.ts

```diff
@@ -533,4 +526,24 @@ export default class DefaultXrpClient implements XrpClientDecorator {
 
     return Utils.toHex(response.getHash_asU8())
   }
+
+  /**
+   * Returns a detailed TransactionResult for a given XRPL transaction hash.
+   *
+   * @param transactionHash The transaction hash to populate a TransactionResult for.
+   * @returns A Promise which resolves to a TransactionResult associated with the given transaction hash.
+   */
+  private async getTransactionResult(
```

Comment:
didn't add tests for this since it's an impl detail of DefaultXrpClient and will be put through its paces by the diff transaction types. lmk though if you think i should test